### PR TITLE
[ORCA-132] Update `cavatica-launch-poc` Airflow DAG to leverage new `orca` package

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
The Airflow DAG is based on the [acceptance test](https://github.com/Sage-Bionetworks-Workflows/py-orca/pull/6/files#diff-262dfd709a9e68a57abb67291f0f05e927acc0747805c45423520e5fbaff731e) that I wrote in `orca`. One issue that I spent a lot of time debugging is that `@task.sensor` doesn't support `params` yet. Fortunately, there is an open issue and pending PR (see comment in DAG for links). Once I solved that problem, the rest was straightforward. 

I still need to make the following modifications once https://github.com/Sage-Bionetworks-Workflows/py-orca/pull/6 is merged and an `orca` release is minted: 

- [x] Remove Git installation in Dockerfile
- [x] Install `orca` from PyPI (not GitHub) and pin major version

Depends on https://github.com/Sage-Bionetworks-Workflows/py-orca/pull/6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,10 @@
 FROM apache/airflow:2.5.0-python3.10
 
+# Install Git for orca package installation (before release)
+USER root
+RUN apt update && apt install git -y
+USER airflow
+
```
Comment (BrunoGrandePhD) [inline Dockerfile:7]: This will be removed once I use the PyPI release of `orca`.

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+markupsafe==2.1.1 ; python_version >= '3.7'
+marshmallow==3.19.0 ; python_version >= '3.7'
+marshmallow-enum==1.5.1
+marshmallow-oneofschema==3.0.1 ; python_version >= '3.6'
+marshmallow-sqlalchemy==0.26.1 ; python_version >= '3.6'
+mdit-py-plugins==0.3.3 ; python_version >= '3.7'
+mdurl==0.1.2 ; python_version >= '3.7'
+numpy==1.23.5 ; python_version >= '3.10'
+oauthlib==3.2.2 ; python_version >= '3.6'
+git+https://github.com/Sage-Bionetworks-Workflows/py-orca@bgrande/ORCA-133/airflow-provider#egg=orca
```
Comment (BrunoGrandePhD) [inline requirements.txt:89]: This will be replaced with a PyPI release.

File: dags/cavatica-launch-poc-v2.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            "outSAMattrRGline": "ID:HCC1187_1M\tLB:Not_Reported\tPL:Illumina\tSM:HCC1187_1M",
+            "output_basename": "cavatica_launch_poc_v2",
+        }
+        clean_run_id = run_id.replace("+00:00", "Z").replace(":", ".")
+        task_id = hook.ops.create_task(clean_run_id, params["app_id"], task_inputs)
+        return task_id
+
+    @task.sensor(poke_interval=30, timeout=604800, mode="poke")
+    def monitor_task(task_name):
+        # TODO: Once the following PR is merged, use `params` argument
```
Comment (thomasyu888) [inline dags/cavatica-launch-poc-v2.py:52]: Nice find! Wow, sensor must be pretty new like you said!

**Review by BWMac — APPROVED** [review]
LGTM

**Review by thomasyu888 — APPROVED** [review]
🛫 LGTM!

**Pull request merged**